subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskExecutionIntegrationTest.groovy

```diff
@@ -603,7 +603,7 @@ task someTask(dependsOn: [someDep, someOtherDep])
         expect:
         2.times {
             succeeds ':build'
-            result.assertTasksExecutedInOrder(':b:jar', ':a:compileJava', ':a:compileFinalizer', ':a:jar', ':build')
+            result.assertTasksExecutedInOrder ':b:jar', ':a:compileJava', any(':a:compileFinalizer', ':a:jar'), ':build'
```

Comment:
I love the fact that `any` means `anyOrder`, as in Hamcrest, it would mean that any of the items satisfy the match. /rant